examples/brainfuck/src/lexer.rs

```diff
@@ -0,0 +1,94 @@
+use super::parser::{Diagnostic, Span};
+use codespan_reporting::diagnostic::Label;
+use logos::Logos;
+
+#[derive(Debug, Clone, PartialEq, Default)]
+pub enum LexerError {
+    #[default]
+    Invalid,
+}
+
+impl LexerError {
+    pub fn into_diagnostic(self, span: Span) -> Diagnostic {
+        match self {
+            Self::Invalid => Diagnostic::error()
+                .with_message("invalid token")
+                .with_labels(vec![Label::primary((), span)]),
+        }
+    }
+}
+
+#[allow(clippy::upper_case_acronyms)]
+#[derive(Logos, Debug, PartialEq, Copy, Clone)]
+#[logos(error = LexerError)]
+pub enum Token {
+    EOF,
+
+    #[token("+")]
+    Plus,
+    #[token("-")]
+    Minus,
+    #[token("<")]
+    Left,
+    #[token(">")]
+    Right,
```

Comment:
I suggest to always use the symbol name for the name of punctuation tokens. In more complicated languages this avoids confusion, when a symbol has different semantic meaning in parts of the grammar.
```suggestion
    #[token("<")]
    Lt,
    #[token(">")]
    Gt,
```